pkg/gather/service/analyze.go

```
@@ -71,6 +74,11 @@ func analyzeGatherBundle(bundleFile io.Reader) error {
 		serviceAnalyses[serviceName] = serviceAnalysis
 	}
 
+	if len(servicesFound) == 0 {
+		logrus.Error("Invalid log bundle or the bootstrap machine could not be reached and bootstrap logs were not collected")
+		return nil
```

Why not return the error here?

---

Because I thought we shouldn't consider this an error state, more like: the analysis ran but there is nothing for it to do.

---

TBH I debated this myself and I'm not sure which one is better UX.

---

This choice is also inline with the previous behavior: even though the message was bogus, the "no files" test case was not considered an error: https://github.com/openshift/installer/pull/8787/files#diff-b06040903f0fe8fc6542ccde27b180fb8817689e7402433d5992733a3eca2448L55-R57

---

I am actually ok with this. Technically the analysis didn't fail, it just had nothing to analyze.